test/models/query_test.rb

```diff
@@ -1160,43 +1161,49 @@ def test_collection_number_all
 
   def test_collection_number_for_observation
     obs = observations(:detailed_unknown_obs)
-    expect = obs.collection_numbers.sort_by(&:format_name)
-    assert_query(expect, :CollectionNumber, observation: obs.id)
+    expects = obs.collection_numbers.sort_by(&:format_name)
+    assert_query(expects, :CollectionNumber, observation: obs.id)
   end
 
   def test_collection_number_pattern_search
-    expect = CollectionNumber.
-             where(CollectionNumber[:name].matches("%Singer%").
-                   or(CollectionNumber[:number].matches("%Singer%"))).
-             sort_by(&:format_name)
-    assert_query(expect, :CollectionNumber, pattern: "Singer")
+    expects = CollectionNumber.
+              where(CollectionNumber[:name].matches("%Singer%").
+                    or(CollectionNumber[:number].matches("%Singer%"))).
+              sort_by(&:format_name)
+    assert_query(expects, :CollectionNumber, pattern: "Singer")
 
-    expect = CollectionNumber.
-             where(CollectionNumber[:name].matches("%123a%").
-                   or(CollectionNumber[:number].matches("%123a%"))).
-             sort_by(&:format_name)
-    assert_query(expect, :CollectionNumber, pattern: "123a")
+    expects = CollectionNumber.
+              where(CollectionNumber[:name].matches("%123a%").
+                    or(CollectionNumber[:number].matches("%123a%"))).
+              sort_by(&:format_name)
+    assert_query(expects, :CollectionNumber, pattern: "123a")
   end
 
   def test_comment_all
-    expect = Comment.all.reverse
-    assert_query(expect, :Comment)
+    expects = Comment.order(Comment[:created_at].desc, Comment[:id].desc).uniq
```

Comment:
I'm curious whether you hit a case where you needed two sort params.
Also about `uniq`. 
Seems like `Comment.order(id: :desc)` should be enough.

Reply:
Never mind.  Looks like you did this to make the `expects` more consistent across different tests.
Makes sense.

Reply:
@JoeCohen I think you can use `distinct` in all test scopes _except_ polymorphic associations. 

For Comment I had to call `uniq` for any of the scopes joining to the `target`, to get it to execute the query right away, or else i got a `polymorphic join cannot be eager loaded` error.

Reply:
Ha! `distinct` works here too.

The current state of the PR is kind of a rolling record of "things I tried", cleaning it up now.